pkg/service/catalog/api/types.go

```diff
@@ -0,0 +1,21 @@
+package api
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+type CatalogService struct {
+	// Target is a reference to a service
+	Target kapi.ObjectReference
+	// Type is a published service's claim type
+	Type string
+	// Description is a user-defined description of the published service
+	Description string
```

Comment:
Annotations probably has to be a prefix or we will have to embed it as
json.  Labels probably has to be a structured selector (or at least be
future compatible with the total syntax).

On Oct 2, 2015, at 3:36 PM, Andy Goldstein <notifications@github.com> wrote:

In pkg/service/catalog/api/types.go
<https://github.com/openshift/origin/pull/4822#discussion_r41059137>:

> @@ -0,0 +1,21 @@
> +package api
> +
> +import (
> +	kapi "k8s.io/kubernetes/pkg/api"
> +)
> +
> +type CatalogService struct {
> +	// Target is a reference to a service
> +	Target kapi.ObjectReference
> +	// Type is a published service's claim type
> +	Type string
> +	// Description is a user-defined description of the published service
> +	Description string

SGTM. Something like this?

openshift.io/servicecatalog: default
openshift.io/servicecatalog.labels: foo=bar,color=red
openshift.io/servicecatalog.displayname: awesomesvc
openshift.io/servicecatalog.description: this is an awesome service
openshift.io/servicecatalog.annotations: a=1,b=2

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4822/files#r41059137>.